nifi-nar-bundles/nifi-pulsar-bundle/nifi-pulsar-processors/src/main/java/org/apache/nifi/processors/pulsar/pubsub/PublishPulsarRecord_1_X.java

```diff
@@ -0,0 +1,294 @@
+import org.apache.pulsar.client.api.Producer;
+
+@Tags({"Apache", "Pulsar", "Record", "csv", "json", "avro", "logs", "Put", "Send", "Message", "PubSub", "1.0"})
+@CapabilityDescription("Sends the contents of a FlowFile as individual records to Apache Pulsar using the Pulsar 1.x client API. "
+    + "The contents of the FlowFile are expected to be record-oriented data that can be read by the configured Record Reader. "
+    + "The complementary NiFi processor for fetching messages is ConsumePulsarRecord_1_0.")
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@WritesAttribute(attribute = "msg.count", description = "The number of messages that were sent to Pulsar for this FlowFile. This attribute is added only to "
+        + "FlowFiles that are routed to success.")
+@SeeAlso({PublishPulsar_1_X.class, ConsumePulsar_1_X.class, ConsumePulsarRecord_1_X.class})
+public class PublishPulsarRecord_1_X extends AbstractPulsarProducerProcessor {
+
+    private static final List<PropertyDescriptor> PROPERTIES;
+    private static final Set<Relationship> RELATIONSHIPS;
+
+    static {
+        final List<PropertyDescriptor> properties = new ArrayList<>();
+        properties.add(PULSAR_CLIENT_SERVICE);
+        properties.add(RECORD_READER);
+        properties.add(RECORD_WRITER);
+        properties.add(TOPIC);
+        properties.add(ASYNC_ENABLED);
+        properties.add(MAX_ASYNC_REQUESTS);
+        properties.add(BATCHING_ENABLED);
+        properties.add(BATCHING_MAX_MESSAGES);
+        properties.add(BATCH_INTERVAL);
+        properties.add(BLOCK_IF_QUEUE_FULL);
+        properties.add(COMPRESSION_TYPE);
+        properties.add(MESSAGE_ROUTING_MODE);
+        properties.add(PENDING_MAX_MESSAGES);
+
+        PROPERTIES = Collections.unmodifiableList(properties);
+
+        final Set<Relationship> relationships = new HashSet<>();
+        relationships.add(REL_SUCCESS);
+        relationships.add(REL_FAILURE);
+        RELATIONSHIPS = Collections.unmodifiableSet(relationships);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+
+        final FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        final String topic = context.getProperty(TOPIC).evaluateAttributeExpressions(flowFile).getValue();
+
+        if (StringUtils.isBlank(topic)) {
+            getLogger().error("Invalid topic specified {}", new Object[] {topic});
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        // Read the contents of the FlowFile into a byte array
+        final byte[] messageContent = new byte[(int) flowFile.getSize()];
```

Comment:
This could break in some use cases. For example, I have processed 50GB text files using NiFi before. Not a common use case, but I'd like to know what others think here.

@bbende @markap14 @alopresto @pvillard31 @mattyb149 can one of you take a look at this and weigh in on whether there is a better way to handle the input here to accommodate very large record sets?